app/src/test/java/com/globant/marvelmvvm/allcharactersTest/AllCharactersViewModelTest.kt

```diff
@@ -0,0 +1,111 @@
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.Assert.assertEquals
+import org.mockito.junit.MockitoJUnitRunner
+import java.lang.Exception
+
+@RunWith(MockitoJUnitRunner::class)
+class AllCharactersViewModelTest {
+
+    @ObsoleteCoroutinesApi
+    private var mainThreadSurrogate = newSingleThreadContext(UI_THREAD)
+
+    @get:Rule
+    val taskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var viewModel: AllCharactersContract.ViewModel
+    private lateinit var responseList: List<Data<out List<Character>>>
+    private var mockedModel: AllCharactersContract.Model = mock()
+    private var validResult: Result.Success<List<Character>> = mock()
+    private var charactersResponse: List<Character> = mock()
+    private var invalidResult: Result.Failure = mock()
+    private var exception: Exception = mock()
+
+    @Before
+    fun setUp(){
+        Dispatchers.setMain(mainThreadSurrogate)
+        viewModel = AllCharactersViewModel(mockedModel)
+    }
+
+    @After
+    fun after() {
+        mainThreadSurrogate.close()
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `when getAllCharacters returns gets error in response`() {
+        val liveDataUnderTest = viewModel.mainState.testObserver()
+        val responseList = listOf(
+            Data(status = Status.LOADING, data = null, error = null),
+            Data(status = Status.RESPONSE_ERROR, data = null, error = exception))
+        whenever(mockedModel.getAllCharacters()).thenReturn(invalidResult)
+        whenever(invalidResult.exception).thenReturn(exception)
+        runBlocking {
+            viewModel.fetchAllCharacters().join()
+        }
+        verify(mockedModel).getAllCharacters()
+        assertEquals(responseList[0].status, liveDataUnderTest.observedValues[0]?.peekContent()?.status)
+        assertEquals(responseList[1].status, liveDataUnderTest.observedValues[1]?.peekContent()?.status)
+        assertEquals(responseList[1].error, liveDataUnderTest.observedValues[1]?.peekContent()?.error)
+    }
+
+    @Test
+    fun `when getAllCharacters returns success response`(){
+        val liveDataUnderTest = viewModel.mainState.testObserver()
+        responseList = listOf(
+            Data(status = Status.LOADING, data = null, error = null),
+            Data(status = Status.RESPONSE_SUCCESS, data = charactersResponse, error = null))
+        whenever(mockedModel.getAllCharacters()).thenReturn(validResult)
+        whenever(validResult.data).thenReturn(charactersResponse)
+        runBlocking {
+            viewModel.fetchAllCharacters().join()
+        }
+        verify(mockedModel).getAllCharacters()
+        assertEquals(responseList[0].status, liveDataUnderTest.observedValues[0]?.peekContent()?.status)
```

Comment:
This it is not really a hardcode, but you added in the componion, two values with the same value...so you can use it

Reply:
Yes, I forgot put them here. Thanks Mau!